### DISCUSSION TITLE
Which tool do you guys use to do wasm size profiling ?

### DISCUSSION THREAD

**Discussion by ravisumit33** [emscripten-core/emscripten#18756] [Q&A] [answered]
Our codebase constantly changes which results in wasm size change. Due to network cost in downloading wasm, we have to keep wasm size at check. We have earlier used [bloaty](https://github.com/google/bloaty) for getting symbols diff between old and new wasm but it looks like bloaty doesn't support wasm binaries perfectly. Bloaty kept giving irrelevant diffs which made us stop using it. We then checked [twiggy](https://github.com/rustwasm/twiggy) but we kept getting error mentioned in https://github.com/rustwasm/twiggy/issues/326 even though our wasm is not multithreaded. What tool do you guys use or recommend for this purpose ?

**Comment by ravisumit33** [comment]
@kripken Any suggestion ?

**Comment by sbc100** [comment, marked as answer]
In emscripten itself we only track overall code size, not the size of individual functions. 

I think bloaty and twiggy are the state of art for tracking size at fine grained level.  If they have issues with wasm or have regressed I think its probably worth opening issues there to get them working again.
